proposals/913-components-repo/README.md

```diff
@@ -0,0 +1,466 @@
+- Components are not published as a Python package today, so consumption requires cloning the monorepo or copying
+  snippets manually.
+
+A dedicated repository with purpose-built tooling will make it easier for users to discover, evaluate, and integrate KFP
+assets while allowing the community to maintain them at its own cadence.
+
+This catalog also creates a durable bridge between Kubeflow Pipelines and the broader Kubeflow ecosystem: it gives the
+SDK and API server a canonical source of reusable building blocks, provides SIGs with a venue to showcase
+Kubernetes-native MLOps patterns, and offers users a distribution channel that evolves independently from the core KFP
+release train. Centralizing community-supported assets in one place keeps the SDK surface area lean while encouraging
+experimentation in third-party areas that can later graduate into core.
+
+### Goals
+
+1. Move reusable components and pipelines into a dedicated GitHub repository with clear structure and governance.
+2. Provide standardized metadata, documentation, and testing requirements for every asset.
+3. Ship an installable Python package for core (community-maintained) components that is versioned to match Kubeflow
+   releases.
+4. Maintain a parallel, clearly demarcated area for third-party contributions, shipped as its own Python package that
+   tracks the same release cadence as the core catalog.
+5. Automate maintenance (e.g. stale component detection, dependency validation) to keep the catalog healthy.
+6. Provide developer onboarding materials and guidance for agents generating components/pipelines.
+
+### Non-Goals
+
+1. Operate turnkey build or publishing pipelines for third-party maintainers. We supply validation tooling and guidance,
+   but external teams continue to manage their own release automation.
+2. Backport historical component versions; only actively maintained assets will be migrated.
+3. Remove the existing `components/` directory in the main KFP repo without notice. We will announce the new repository,
+   point contributors at the new onboarding documentation, and give the community a minimum of 90 days to move their
+   assets before deleting the legacy directory.
+4. Vendor specific components. Vendors are highly encouraged to use the same repository structure in their own
+   repository.
+
+## Proposal
+
+### Repository Layout
+
+Create a new repository (e.g. `kubeflow/kfp-components`) with the following top-level layout:
+
+```text
+root
+├── components
+│   ├── README.md (catalog landing page)
+│   ├── __init__.py (auto-imports all components for clean `from kfp_components import training` usage)
+│   ├── training/
+│   │   ├── README.md (category index listing each component with summaries/links)
+│   │   ├── __init__.py (re-exports all components in this category)
+│   │   └── <component-name>/
+│   │       ├── __init__.py (exposes the component entrypoint for imports)
+│   │       ├── component.py
+│   │       ├── metadata.yaml
```

Comment:
As discussed in the community meeting, we need to consider how we will manage the docker images (which are part of the component).

My preference is that we require components to ONLY use either:
1. an approved "base" docker image
2. an extension of one of the "approved base images" using a Dockerfile defined in this repo (possibly under the components folder).